release-22.1: sql,server: increase severity of upgraded-related logging

knz opened the pull request:
Backport 1/1 commits from #90165.
Backport 1/1 commits from #91049. 

/cc @cockroachdb/release

---

Informs #90148.

This increases the severity from INFO in the following cases:

- in the case when `SET CLUSTER SETTING version` is issued from a SQL client (WARNING in case of failure).
- in the case when the server spontaneously decides to upgrade in the background (ERROR in case of failure).

Release note: None

---
Release justification: increases troubleshootability

knz committed:
sql,server: increase severity of upgraded-related logging

This increases the severity from INFO in the following cases:

- in the case when `SET CLUSTER SETTING version` is issued from a SQL
  client (WARNING in case of failure).
- in the case when the server spontaneously decides to upgrade
  in the background (ERROR in case of failure).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/90234)
<!-- Reviewable:end -->

knz committed:
server: clean up startAttemptUpgrade

This avoids a log error when `preserve_downgrade_option` is set, and
overall simplifies the control flow.

It also avoids redaction of the version number.

Release note: None

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/90234#-:-NFnhXUf-Gtw2wSUKt1p:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

knz commented:
we'll hold on on this as we also need to backport #91049.

knz commented:
> we'll hold on on this as we also need to backport https://github.com/cockroachdb/cockroach/pull/91049.
Done

rafiss reviewed (approved):
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/90234)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->
